api/mysagw/case/serializers.py

```diff
@@ -115,11 +128,11 @@ def create(self, validated_data):
         return new_accesses
 
     def _send_mail(self, new_accesses_for_user):
-        identity = new_accesses_for_user[0].identity
+        identity = new_accesses_for_user[0][0].identity
         subject = case_transfer.EMAIL_BULK_INVITE_SUBJECTS[identity.language]
         links = "\n".join(
             [
-                f"{settings.SELF_URI}/cases/{access.case_id}"
+                f"{access[1]} - {settings.SELF_URI}/cases/{access[0].case_id}"
                 for access in new_accesses_for_user
```

Comment:
And on a more philosophical note: isn't that a somewhat unusual place for a side effect like sending mail on creating a db entry? It's maybe a matter of taste.

Reply:
IMO the serializer is the right place for this. Where would you put it?

One could argue, that this whole endpoint is not restful at all. But since we want to be able to do this with one single request, no matter how many cases are involved, I've decided to go this route.

Reply:
Before I knew that you have no access to the DB I'd have said in a `signal`. So in this case I'd probably do it in the view to keep businesslogic out of the serializer.

Reply:
IMO the view is only for handling the requests. My approach to business logic is in that order:

1. If it's concerning a model instance, it goes to the model.
2. if it needs to be done before any serializer validation, it goes into the view (e.g. views, that need to specially handle headers). That's very rare.
3. If it needs to be called in different ways, it goes to it's own module and gets called in the serializer. That's what we did in Caluma. Before, we also had it in the serializer.
4. If it's overly complex, it also goes into it's own module and gets called in the serializer.
5. Everything else goes into the serializer.